src/borg/archive.py

```diff
@@ -1976,10 +1977,26 @@ def save(self, archive, target, comment=None, replace_original=True):
             return
         if comment is None:
             comment = archive.metadata.get('comment', '')
-        target.save(comment=comment, stats=target.stats, additional_metadata={
+        duration = timedelta(seconds=time.monotonic() - target.start_monotonic)
+
+        # Keep for the statistics if necessary
+        if self.stats:
+            _start = target.start
+
+        if self.timestamp is None:
+            end = archive.metadata.get('time_end') or archive.metadata.time
+            start = archive.metadata.time
+        else:
+            end = (self.timestamp + duration).strftime(ISO_FORMAT)
+            start = self.timestamp.strftime(ISO_FORMAT)
```

Comment:
looks like `target.save` method is doing just that when it was given a timestamp != None as parameter.

so guess we only need to prepare a different `additional_metadata` dict depending on timestamp being None or not:
- None: dict with start/end info of original archive (see that comment) to override what is normally computed
- not None: dict without start/end overrides